src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -39,59 +33,94 @@ const propTypes = {
     A positive value shifts the tooltip down, and a negative value shifts it up. */
     shiftVertical: PropTypes.number.isRequired,
 
-    /** Callback to set the Ref to the Tooltip */
-    setTooltipRef: PropTypes.func.isRequired,
-
     /** Text to be shown in the tooltip */
     text: PropTypes.string.isRequired,
 
-    /** Callback to be used to calulate the width and height of tooltip */
-    measureTooltip: PropTypes.func.isRequired,
-};
+    /** Number of pixels to set max-width on tooltip  */
+    maxWidth: PropTypes.number.isRequired,
 
-const defaultProps = {};
-
-const TooltipRenderedOnPageBody = (props) => {
-    const {
-        animationStyle,
-        tooltipWrapperStyle,
-        tooltipTextStyle,
-        pointerWrapperStyle,
-        pointerStyle,
-    } = getTooltipStyles(
-        props.animation,
-        props.windowWidth,
-        props.xOffset,
-        props.yOffset,
-        props.wrapperWidth,
-        props.wrapperHeight,
-        props.tooltipWidth,
-        props.tooltipHeight,
-        props.shiftHorizontal,
-        props.shiftVertical,
-    );
-    return ReactDOM.createPortal(
-        <Animated.View
-            ref={props.setTooltipRef}
-            onLayout={props.measureTooltip}
-            style={[tooltipWrapperStyle, animationStyle]}
-        >
-            <Text style={tooltipTextStyle} numberOfLines={1}>{props.text}</Text>
-            <View style={pointerWrapperStyle}>
-                <View style={pointerStyle} />
-            </View>
-        </Animated.View>,
-        document.querySelector('body'),
-    );
+    /** Maximum number of lines to set on tooltip */
+    numberOfLines: PropTypes.number.isRequired,
 };
 
-TooltipRenderedOnPageBody.propTypes = propTypes;
-TooltipRenderedOnPageBody.defaultProps = defaultProps;
-TooltipRenderedOnPageBody.displayName = 'TooltipRenderedOnPageBody';
-
 // Props will change frequently.
 // On every tooltip hover, we update the position in state which will result in re-rendering.
 // We also update the state on layout changes which will be triggered often.
 // There will be n number of tooltip components in the page.
 // Its good to memorize this one.
-export default memo(TooltipRenderedOnPageBody);
+class TooltipRenderedOnPageBody extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            // The width of toolttip's inner text
+            tooltipTextWidth: 0,
+
+            // The width and height of the tooltip itself
+            tooltipWidth: 0,
+            tooltipHeight: 0,
+        };
+
+        this.measureTooltip = this.measureTooltip.bind(this);
+    }
+
+    componentDidMount() {
+        this.textRef.measure((x, y, width) => {
+            this.setState({
+                tooltipTextWidth: width,
+            });
+        });
+    }
```

Comment:
I was not saying to use the measure call here. I was thinking that we can use measure where we need the `tooltipTextWidth`. Like in the getTooltipStyles calculations. But I understand why do you need to save the width in `componentDidMount`. Using measure here is not useful at all. The previous OffsetWidth attempt was also fine. I would say it was better as it was quick. Now, this measure call is adding the async code which will further add delay. 

---
Please feel free to ask for confirmation before making changes, if you don't understand requests. It will save our time.

Reply:
Sorry for premature pushed changes, I understand now your point of view. I tried using `measure` inside `getTooltipStyles` but I couldn't make it work  so I am reverting to `offsetWidth` solution. (along with comments changes of course).